src/main/java/com/limechain/network/Network.java

```diff
@@ -138,6 +149,20 @@ private void initializeProtocols(ChainService chainService, HostConfig hostConfi
         kademliaService.setHost(host);
     }
 
+    private Ed25519PrivateKey loadPrivateKeyFromDB(KVRepository<String, Object> repository) {
+        final Ed25519PrivateKey privateKey;
+        Optional<Object> peerIdKeyBytes = repository.find(this.peerIdKey);
+        if(peerIdKeyBytes.isPresent()){
+            privateKey = Ed25519Utils.loadPrivateKey((byte[]) peerIdKeyBytes.get());
+            log.log(Level.INFO, "PeerId loaded from database!");
+        }else {
+            privateKey = Ed25519Utils.generatePrivateKey();
+            repository.save(this.peerIdKey, privateKey.raw());
+            log.log(Level.INFO, "Generated new peerId ");
+        }
+        return privateKey;
+    }
+
```

Comment:
Add some unit tests for this functionality and please check if the other tests need to be adjusted or extended.